src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/ScanBlobScanLogHybridPollingStrategy.cs

```diff
@@ -0,0 +1,162 @@
+                _registrations.Add(container, containerRegistrations);
+            }
+
+            containerRegistrations.Add(triggerExecutor);
+            return Task.FromResult(0);
+        }
+
+        public async Task<TaskSeriesCommandResult> ExecuteAsync(CancellationToken cancellationToken)
+        {
+            ThrowIfDisposed();
+            await Task.WhenAll(_scanContainerStrategy.ExecuteAsync(cancellationToken),
+                                _pollLogStrategy.ExecuteAsync(cancellationToken));
+
+            // Run subsequent iterations at 2 second intervals.
+            return new TaskSeriesCommandResult(wait: Task.Delay(TwoSeconds));
+        }
+
+        public void Notify(IStorageBlob blobWritten)
+        {
+            ThrowIfDisposed();
+            _scanContainerStrategy.Notify(blobWritten);
+            _pollLogStrategy.Notify(blobWritten);
+        }
+
+        public void Dispose()
+        {
+            if (!_disposed)
+            {
+                _pollLogStrategy.Dispose();
+                _scanContainerStrategy.Dispose();
+                _cancellationTokenSource.Dispose();
+                _disposed = true;
+            }
+        }
+
+        private void ScnaContainersAndRegisterWithStrategy(CancellationToken cancellationToken)
+        {
+            foreach (IStorageBlobContainer container in _registrations.Keys)
+            {
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    return;
+                }
+
+                IStorageBlobResultSegment results;
+
+                try
+                {
+                    // Non-async is correct here. ScanContainers occurs on a background thread. Unless it blocks, no one
+                    // else is around to observe the results.
+                    results = container.ListBlobsSegmentedAsync(prefix: null, useFlatBlobListing: true, blobListingDetails: WindowsAzure.Storage.Blob.BlobListingDetails.None, maxResults: _scanBlobThreshold, currentToken: null, options: null, operationContext: null,
+                        cancellationToken: CancellationToken.None).GetAwaiter().GetResult();
+                }
+                catch (StorageException exception)
+                {
+                    if (exception.IsNotFound())
+                    {
+                        return;
+                    }
+                    else
+                    {
+                        throw;
+                    }
+                }
+
+                // decide which strategy to use based on the existance of continuation token
+                // if one exists it means there are more use PollLogStrategy otherwise use 
+                // ScanBlobStrategy
+
+                if (results.ContinuationToken != null)
+                {
+                    _pollStrategies.Add(container,PollStrategy.ScanLogs);
+                    foreach (var triggerExecuter in _registrations[container])
```

Comment:
This entire foreach loop is duplicated below. Hoist it out of the if/else and modify it to operate on an IBlobListenerStrategy local that you set in the if/else